### PR TITLE
fix(sdk): fix boolean default value serialization bug

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Fixes a bug where default values for boolean inputs were sometimes 
serialized as numbers (e.g., `1.0` for `true`) instead of booleans.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title 
convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/test_data/pipelines/xgboost_sample_pipeline.yaml [hunk 1]
```diff
@@ -132,19 +132,19 @@ components:
           defaultValue: gbtree
           parameterType: STRING
         label_column:
-          defaultValue: '0'
+          defaultValue: 0.0
```
Comment (connor-mccarthy) [inline sdk/python/test_data/pipelines/xgboost_sample_pipeline.yaml:135]: Note that these changes were not the intend of the bug, but the improved `default_value` setting logic caught this type inconsistency that was previously bypassed.

**Comment by connor-mccarthy** [comment]
[Test failures](https://github.com/kubeflow/pipelines/pull/8444#issuecomment-1312219101) are unrelated to these changes

**Comment by IronPan** [comment]
/lgtm
Thanks for the quick fix!

**Comment by connor-mccarthy** [comment]
Manually merging, as test failures are unrelated to this PR.

I tested execution of sample pipelines with the corrected `true`/`false` boolean defaults, as well as the xgboost pipeline sample that was updated by this PR.

**Comment by connor-mccarthy** [comment]
Fixes #8303

**Comment by connor-mccarthy** [comment]
Fixes #8302

**Pull request merged**